file/file_client.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/golang/protobuf/proto"
+	monitoring "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// FileClient allows writing to a file gRPC endpoint. The
+// implementation may hit a single backend, so the application should create a
+// number of these clients.
+type FileClient struct {
+	logger log.Logger
+	file *os.File
+}
+
+// NewFileClient creates a file under os.TempDir(), and creates a new FileClient writing to
+// the file. The user of NewFileClient is responsible to manage the created file.
+func NewFileClient(logger log.Logger) *FileClient {
+        if logger == nil {
+                logger = log.NewNopLogger()
+        }
+	tmpOutputDir := os.TempDir() + "/stackdriver-prometheus-sidecar/CreateTimeSeriesRequest"
```

Comment:
Can you pass the file in to make testing easier and not require a file system?

You can always default to this code if the file is `nil`, though personally I'd leave it out. It's more conventional for programs to accept a file name on the input instead of creating a temp one to return data; in some implementation the tempdir should actually get deleted when the process ends. I'm fine not pushing this code out in this PR, if you don't need it.

Reply:
Given your feedback, I think the file creation should be moved from Client itself to Factory method. Client itself only accepts the io.WriteCloser. Therefore, I restructure the boundary between factory and client to reflect the change.